hyperspy/roi.py

```diff
@@ -322,11 +322,13 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.events.data_changed.connected:
+            signal.events.data_changed.connect(self.update, [])
```

Comment:
Now that I think about it, I should probably change the example and have both of the rois in the original image (since it defines the data that's being sampled), however add line widget to both plots (to simulate a "zoom")... Though still not sure it makes sense as an example to begin with. Any suggestions?